File: research_materials/simulation/project_paths.py
from __future__ import annotations

from pathlib import Path


ROOT = Path(__file__).resolve().parent

RUN = ROOT / "run"
EXPORT = ROOT / "export"
PLOT = ROOT / "plot"
SRC = ROOT / "src"

DATA = ROOT / "data"
RAW = DATA / "raw"
RAW_DYMOLA = RAW / "dymola"
PROCESSED = DATA / "processed"
STAGE1 = DATA / "stage1"
GAUSSIAN = STAGE1 / "gaussian"
REGD = DATA / "regd"
BASELINE = DATA / "baseline"
RESERVE = DATA / "reserve"
POSTPROCESS = DATA / "postprocess"
POSTPROCESS_MARKET = POSTPROCESS / "market"
POSTPROCESS_COST = POSTPROCESS / "cost"
EXPORTS = DATA / "exports"
FIGURES = DATA / "figures"

SUPPLEMENT_ROOT = ROOT.with_name(f"{ROOT.name}_supplement")
ARCHIVE_ROOT = ROOT.with_name(f"{ROOT.name}_archive")
SUPPLEMENT_STAGE1_TEST_DAYS = SUPPLEMENT_ROOT / "data" / "stage1" / "test_days"
SUPPLEMENT_MULTIDAY_BETA90_NSCAN20 = SUPPLEMENT_ROOT / "data" / "multiday_beta90_nscan20"
SUPPLEMENT_EXPORTS_MULTIDAY = SUPPLEMENT_ROOT / "data" / "exports" / "07_multiday_robustness_beta90_nscan20"


def data_file(area: str, *parts: str) -> Path:
    key = area.replace("-", "_").lower()
    areas = {
        "raw": RAW,
        "raw_dymola": RAW_DYMOLA,
        "dymola": RAW_DYMOLA,
        "processed": PROCESSED,
        "processeddata": PROCESSED,
        "stage1": STAGE1,
        "stage1data": STAGE1,
        "gaussian": GAUSSIAN,
        "rousseaugaussian": GAUSSIAN,
        "regd": REGD,
        "baseline": BASELINE,
        "baseline_fixedts": BASELINE,
        "reserve": RESERVE,
        "reservecostcurve_hourly": RESERVE,
        "postprocess_market": POSTPROCESS_MARKET,
        "market": POSTPROCESS_MARKET,
        "marketview": POSTPROCESS_MARKET,
        "postprocess_cost": POSTPROCESS_COST,
        "cost": POSTPROCESS_COST,
        "costview": POSTPROCESS_COST,
        "exports": EXPORTS,
        "export": EXPORTS,
        "figures": FIGURES,
    }
    try:
        base = areas[key]
    except KeyError as exc:
        raise KeyError(f"Unknown data area: {area}") from exc
    return base.joinpath(*parts)

File: research_materials/simulation/project_paths.m
function P = project_paths(rootDir)
%PROJECT_PATHS Central path registry for the paper simulation cleanroom.
%
%   P = project_paths() returns absolute paths rooted at this cleanroom.
%   P = project_paths(rootDir) uses the supplied root instead.

if nargin < 1 || isempty(rootDir)
    rootDir = fileparts(mfilename('fullpath'));
end

P = struct();
P.root = rootDir;

P.run = fullfile(rootDir, 'run');
P.export = fullfile(rootDir, 'export');
P.plot = fullfile(rootDir, 'plot');
P.src = fullfile(rootDir, 'src');

P.data = fullfile(rootDir, 'data');
P.raw = fullfile(P.data, 'raw');
P.raw_dymola = fullfile(P.raw, 'dymola');
P.processed = fullfile(P.data, 'processed');
P.stage1 = fullfile(P.data, 'stage1');
P.gaussian = fullfile(P.stage1, 'gaussian');
P.regd = fullfile(P.data, 'regd');
P.baseline = fullfile(P.data, 'baseline');
P.reserve = fullfile(P.data, 'reserve');
P.postprocess = fullfile(P.data, 'postprocess');
P.postprocess_market = fullfile(P.postprocess, 'market');
P.postprocess_cost = fullfile(P.postprocess, 'cost');
P.exports = fullfile(P.data, 'exports');
P.figures = fullfile(P.data, 'figures');

P.docs = fullfile(rootDir, 'docs');
P.paper = fullfile(rootDir, 'paper');
P.origin = fullfile(rootDir, 'origin');

P.supplement_root = fullfile(fileparts(rootDir), [get_last_path_part(rootDir) '_supplement']);
P.archive_root = fullfile(fileparts(rootDir), [get_last_path_part(rootDir) '_archive']);
P.supplement_stage1_test_days = fullfile(P.supplement_root, 'data', 'stage1', 'test_days');
P.supplement_multiday_beta90_nscan20 = fullfile(P.supplement_root, 'data', 'multiday_beta90_nscan20');
P.supplement_exports_multiday = fullfile(P.supplement_root, 'data', 'exports', '07_multiday_robustness_beta90_nscan20');
end

function name = get_last_path_part(pathStr)
[~, name] = fileparts(pathStr);
end
